gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -2786,3 +2801,126 @@ gst_tensor_filter_check_hw_availability (const gchar * name, const accl_hw hw)
 
   return available;
 }
+
+/* extern functions for shared model representation table */
+/**
+ * @brief Insert the new shared model representation at the table.
+ * @param[in] key The new key to store at hash table.
+ * @param[in] value The new value to store at hash table.
+ * @param[in] instance The instance that is sharing the model representation. It will be registered at the referred list.
+ * @return 0 if the new key and value are inserted. Others are failed to insert it.
+ */
+int
+nnstreamer_filter_shared_table_insert (char *key, void *value, void *instance)
+{
```

Comment:
This function is not thread safe (against remove function). Cover critical section with mutex.